core/domain/rights_manager_test.py

```diff
@@ -835,102 +647,56 @@ def test_setting_rights_of_collection(self):
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_a).can_unpublish(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_access_activity(
+            self.user_b, collection_rights))
+        self.assertFalse(rights_manager.check_can_unpublish_activity(
+            self.user_a, collection_rights))
 
         rights_manager.unpublish_collection(
             self.user_id_admin, self.COLLECTION_ID)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_a).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_a).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_access_activity(
+            self.user_a, collection_rights))
+        self.assertFalse(rights_manager.check_can_access_activity(
+            self.user_b, collection_rights))
 
     def test_can_only_delete_unpublished_collections(self):
         self.save_new_default_collection(self.COLLECTION_ID, self.user_id_a)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_a).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_delete_activity(
+            self.user_a, collection_rights))
 
         rights_manager.publish_collection(self.user_id_a, self.COLLECTION_ID)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_a).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertFalse(rights_manager.check_can_delete_activity(
+            self.user_a, collection_rights))
 
         rights_manager.unpublish_collection(
             self.user_id_admin, self.COLLECTION_ID)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_a).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-
-
-class CheckCanUnpublishExplorationTest(test_utils.GenericTestBase):
```

Comment:
Now, there is a common function for activity and the tests is shifted to CheckCanUnpublishActivityTest.